.github/workflows/on-pull-request.yml

```diff
@@ -44,29 +50,27 @@ jobs:
       - uses: Swatinem/rust-cache@v2
 
       - name: Build
-        run: cargo build --verbose
-      - name: Cargo Check
-        run: cargo check
+        run: make build
       - name: Test Setup
         # This script relies on dev dependencies such as Tokio, so we run it with the --example flag
         run: cargo run --example test-setup
-      - name: Unit Tests
-        run: cargo test --lib
-      - name: Doc Tests
-        run: cargo test --doc
+      - name: Unit tests
+        run: make test-unit
       - name: Integration Tests
-        run: cargo test --tests
+        run: make test-integration
+      - name: Doc Tests
+        run: make test-doctests
       - name: Test Teardown
         # We want the teardown to execute even if an earlier step fails, hence the `always()` condition
         if: always()
         # This script relies on dev dependencies such as Tokio, so we run it with the --example flag
         run: cargo run --example test-teardown
 
   rustfmt-build-examples:
-    name: Style & Lint
+    name: examples - Style & Lint
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```

Comment:
Should we upgrade the other uses of this to v3?

Reply:
sure ty